.github/workflows/deploy_branch_npm.reusable.workflow.yml

```diff
@@ -0,0 +1,180 @@
+# Reusable workflow for building an application with npm
+# Some steps can be enabled/disabled with workflow parameters
+#
+name: "Build npm"
+
+on:
+  # Reusable workflow
+  workflow_call:
+    inputs:
+      docker-tag-push:
+        description: 'Whether to tag and push docker image to Docker registry'
+        required: false
+        type: boolean
+        default: false
+      package-json-file:
+        description: The path to project's package.json
+        required: false
+        default: 'package.json'
+        type: string
+      docker-file:
+        description: |
+          The path of the Dockerfile to be built.
+          Path can be either a relative path, in which case it is relative to the project's
+          workspace directory (e.g. './path/to/Dockerfile'), or an aboslute path
+          (e.g. '${GITHUB_WORKSPACE}/path/to/Dockerfile')
+          Note that there GitHub workflows enforce limitations for accessing files
```

Comment:
beginning of this sentence does not make sense...one of following would work:
`Note that there are Github workflows which enforce...`
or 
`Note that Github workflows enforce...`